Rename macros and build options NAMEDTENSOR_ENABLED to BUILD_NAMEDTENSOR

xuhdev opened the pull request:
Currently the build system accepts USE_NAMEDTENSOR from the environment
variable and turns it into NAMEDTENSOR_ENABLED when passing to CMake.
This discrepancy does not seem necessary and complicates the build
system. The naming of this build option is also semantically incorrect
("BUILD_" vis-a-vis "USE_").  This commit eradicate this issue before it
is made into a stable release.

The support of NO_NAMEDTENSOR is also removed, since PyTorch has been
quite inconsistent about "NO_*" build options.

---

Note: All environment variables with their names starting with `BUILD_` are currently automatically passed to CMake with no need of an additional wrapper.

xuhdev commented:
Besides all the renaming changes, three lines in `tools/setup_helpers/cmake.py` and `tools/setup_helpers/env.py` that do build option conversion are deleted.

ezyang commented:
@zou3519 can you check on this and approve if you think it is good?

zou3519 commented:
@xuhdev could you please add `[namedtensor ci]` to your commit message so that the named tensor CI job runs on this pull request?

zou3519 commented:
@xuhdev I landed a few things this morning that added some more uses of the macro, so you'll need to rebase. I'll hold off on landing more changes for a day or two so that we can get this PR in.

ezyang commented:
@zou3519 feel free to commandeer this patchset if you think that would be easier

xuhdev commented:
@zou3519 rebased and added the ci command

zou3519 reviewed (approved):
Looks fine, just waiting for tests to pass

zou3519 commented on aten/src/THC/generic/THCTensorMathPointwise.cu:
```diff
@@ -321,9 +322,6 @@ void THCTensor_(sigmoid)(THCState* state, THCTensor* self_, THCTensor* src) {
   }
 
   THCudaCheck(cudaGetLastError());
-#ifdef NAMEDTENSOR_ENABLED
```
This is wrong; I think you intended to change NAMEDTENSOR_ENABLED to BUILD_NAMEDTENSOR.

xuhdev replied:
Right. I probably messed it up during conflict resolving.

zou3519 commented on aten/src/THC/generic/THCTensorMathPointwise.cu:
```diff
@@ -336,9 +334,6 @@ void THCTensor_(digamma)(THCState* state, THCTensor* self_, THCTensor* src) {
   }
 
   THCudaCheck(cudaGetLastError());
-#ifdef NAMEDTENSOR_ENABLED
```
same here

zou3519 commented on aten/src/THC/generic/THCTensorMathPointwise.cu:
```diff
@@ -362,9 +357,6 @@ void THCTensor_(polygamma)(THCState* state, THCTensor* self_, int64_t n, THCTens
   }
 
   THCudaCheck(cudaGetLastError());
-#ifdef NAMEDTENSOR_ENABLED
```
same here

xuhdev committed:
Rename macros and build options NAMEDTENSOR_ENABLED to BUILD_NAMEDTENSOR

Currently the build system accepts USE_NAMEDTENSOR from the environment
variable and turns it into NAMEDTENSOR_ENABLED when passing to CMake.
This discrepancy does not seem necessary and complicates the build
system. The naming of this build option is also semantically incorrect
("BUILD_" vis-a-vis "USE_").  This commit eradicate this issue before it
is made into a stable release.

The support of NO_NAMEDTENSOR is also removed, since PyTorch has been
quite inconsistent about "NO_*" build options.

[namedtensor ci]
